safe_transaction_service/history/serializers.py

```diff
@@ -241,12 +241,16 @@ def save(self, **kwargs):
         safe_address = self.context['safe_address']
         ethereum_client = EthereumClientProvider()
         safe = Safe(safe_address, ethereum_client)
-        try:
-            safe_tx_gas = safe.estimate_tx_gas(self.validated_data['to'], self.validated_data['value'],
-                                               self.validated_data['data'], self.validated_data['operation'])
-        except IOError as exc:
-            raise NodeConnectionException(f'Node connection error when estimating gas for safe {safe_address}') from exc
-        return {'safe_tx_gas': safe_tx_gas}
+        exc = None
+        # Retry twice to get an estimation
+        for _ in range(2):
```

Comment:
Maybe if you need to retry twice, it would be better to do it in the `expect` block where you would handle the first failure 🤔 It would add a lot of nesting though, and the result is the same but yeah, maybe it would be more explicit.

Reply:
The advantage of this, is that you could log the error once and log that you are "retrying" in case that is helpful

Reply:
Good suggestion, but I don't want to log as the error is due to using shared nodes and them being busy when asking for the gas estimation 😉